docs/api.md

```diff
@@ -0,0 +1,248 @@
+# API documentation
+
+- [Nomenclature](#nomenclature)
+- [Requirements for API consumers](#requirements-for-api-consumers)
+  - [Pass the Session Identifier to Access Restricted endpoints](#pass-the-session-identifier-to-access-restricted-endpoints)
+  - [Update the user's session if a new Session Identifier is returned](#update-the-users-session-if-a-new-session-identifier-is-returned)
+  - [End the user's session if an Access Restricted endpoint returns a `401: Unauthenticated`](#end-the-users-session-if-an-access-restricted-endpoint-returns-a-401-unauthenticated)
+  - [Ensure responses are not cached](#ensure-responses-are-not-cached)
+- [API endpoints](#api-endpoints)
+  - [`GET /api/oauth2/sign-in`](#get-apioauth2sign-in)
+  - [`POST /api/oauth2/callback`](#post-apioauth2callback)
+  - [`POST /api/oauth2/state`](#post-apioauth2state)
+  - [`GET /api/attributes`](#get-apiattributes)
+  - [`PATCH /api/attributes`](#patch-apiattributes)
+  - [`GET /api/transition-checker-email-subscription`](#get-apitransition-checker-email-subscription)
+  - [`POST /api/transition-checker-email-subscription`](#post-apitransition-checker-email-subscription)
+- [Healthcheck](#healthcheck)
+
+## Nomenclature
```

Comment:
Can we use a more plain English word, for example, definitions or defined terms?

Reply:
Maybe, but "nomenclature" is what other GOV.UK projects are using, eg:

- https://github.com/alphagov/email-alert-api#nomenclature
- https://github.com/alphagov/content-publisher/#nomenclature
- https://github.com/alphagov/support-api/#nomenclature
- https://github.com/alphagov/finder-frontend/#nomenclature

Reply:
I'll see if I can rework the rest of the doc to not need custom terminology